<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="robot" default="marv" doc="Robot type: husky, tradr or marv"/>
    <arg name="lss_config_path" default="$(dirname)/../../monoforce/config/lss_cfg.yaml"/>
    <arg name="weights" default="$(dirname)/../../monoforce/config/weights/lss/lss.pt"/>
    <arg name="img_topics" default="['/camera_front/image_color/compressed',
                                     '/camera_left/image_color/compressed',
                                     '/camera_rear/image_color/compressed',
                                     '/camera_right/image_color/compressed']"/>
    <arg name="camera_info_topics" default="['/camera_front/camera_info',
                                             '/camera_left/camera_info',
                                             '/camera_rear/camera_info',
                                             '/camera_right/camera_info']"/>
    <!-- dphysics arguments -->
    <arg name="engine" default="torch" doc="Using NVIDIA Warp- or Pytorch-based differentiable physics"/>
    <arg name="device" default="cuda" doc="Device to run the differentiable physics on"/>

    <!-- RGB Terrain Encoder -->
    <include file="$(dirname)/terrain_encoder.launch">
        <arg name="img_topics" value="$(arg img_topics)"/>
        <arg name="camera_info_topics" value="$(arg camera_info_topics)"/>
        <arg name="output_topic" value="height_map"/>
        <arg name="robot_frame" value="base_link"/>
        <arg name="max_msgs_delay" value="0.2"/>
        <arg name="weights" value="$(arg weights)"/>
        <arg name="lss_config_path" value="$(arg lss_config_path)"/>
    </include>

    <!-- Diff Physics: paths sampling -->
    <include file="$(dirname)/dphysics.launch">
        <arg name="engine" value="$(arg engine)"/>
        <arg name="robot" value="$(arg robot)"/>
        <arg name="gridmap_topic" value="grid_map/terrain"/>
        <arg name="gridmap_frame" value="base_link"/>
        <arg name="device" value="$(arg device)"/>
    </include>
</launch>
